microprofile/WFLY-12342_server_readiness_probes.adoc

```diff
@@ -0,0 +1,95 @@
+= [WFLY-12342] Integrate server probes in MP Health readiness check
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,health,openshift
+
+== Overview
+
+MicroProfile Health 2.0 introduced a readiness health check.
+WFLY-12228 added support for MicroProfile Health 2.0 without enhancing the readiness health check.
+
+This means that the readiness probe always return UP by default (as long as the microprofile-health-smallrye is installed).
+
+This RFE enhances it with 3 probes (that are part of the https://github.com/jboss-container-images/jboss-eap-modules/blob/251f422c97e1bcd3625f57295bc79973193482a6/os-eap-probes/2.0/added/probes/probe/eap/dmr.py#L23[EAP readiness probe script]):
```

Comment:
@jmesnil I assume implementation of probes is different. Or is there chance they reuse shared code?

Reply:
The probes that check server states delegates to internal DMR operations (see https://github.com/wildfly/wildfly/pull/13397/files#diff-ca09758a51bffeb0751512740f73f2ccR27). So they rely on the same implementation code.

Reply:
Good. But still differences could be in environment properties driving the logic e.g. [1] and output format?

[1] https://github.com/jboss-container-images/jboss-eap-modules/blob/251f422c97e1bcd3625f57295bc79973193482a6/os-eap-probes/2.0/added/probes/probe/eap/dmr.py#L79

Reply:
Thinking about significance of output similarities of both ways. In case of Kubernetes it is just return code of call important [1] In case of shell probe it is 0 for success and in case of http call it is 200 for success.

So I assume as far as it is true that /health `UP` => 200 and `DOWN` => 50x and it maps to pyton probe logic `UP` => 0 and `DOWN` => 1, we should be safe.

Regarding env variable `PROBE_DISABLE_BOOT_ERRORS_CHECK`. Not sure if we want to copy this to /health implementation.  This is not documented to users and seems to me not consistent to have only one disable check from 3 checks. But If I am reading this correctly it is true by default in our images, so seems it has its meaning? [2]

[1] https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-command
[2] https://github.com/jboss-container-images/jboss-eap-modules/blob/master/os-eap-probes/common/module.yaml